modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1307,11 +1307,14 @@ private void _validateAccountEntryRestrictedObjectFieldId(
 		}
 
 		if (objectDefinition.isApproved() &&
-			objectDefinition.isAccountEntryRestricted()) {
+			((accountEntryRestricted !=
+				objectDefinition.isAccountEntryRestricted()) ||
+			 (accountEntryRestrictedObjectFieldId !=
+				 objectDefinition.getAccountEntryRestrictedObjectFieldId()))) {
 
 			throw new ObjectDefinitionAccountEntryRestrictedException(
-				"Account entry restriction cannot be disabled when the " +
-					"object definition is published");
+				"Account entry restriction cannot be altered when the object " +
+					"definition is published");
```

Comment:
Thanks, @TamyrisTorres. In this case, you must also change the UI, by disabling the `accountEntryRestricted` and `accountEntryRestrictedObjectFieldId` fields [here](https://github.com/liferay/liferay-portal/blob/773a927c59f18efadd6c3c137bb61b3e8abb76d8/modules/apps/object/object-web/src/main/resources/META-INF/resources/object_definitions/object_definition/details.jsp#L205-L212).

Right now the, the `accountEntryRestricted` and `accountEntryRestrictedObjectFieldId` fields are disabled if:
 ```
 objectDefinition.isAccountEntryRestricted() && objectDefinition.isApproved()
 ```
 
 But if they must be disabled whenever the object is published, we must remove the  `objectDefinition.isAccountEntryRestricted()` from the condition.

Reply:
In addition, please change from `altered` to `changed`